wagtail/documents/views/chooser.py

```diff
@@ -50,9 +50,15 @@ def chooser(request):
     q = None
     if 'q' in request.GET or 'p' in request.GET or 'collection_id' in request.GET:
 
+        # Filter by collection
+        current_collection = None
         collection_id = request.GET.get('collection_id')
         if collection_id:
-            documents = documents.filter(collection=collection_id)
+            try:
+                current_collection = Collection.objects.get(id=collection_id)
```

Comment:
No need to fetch the current collection. We only use it to change the message in the template, so we can just check whether `collection_id` is in the GET request. I know this will not check if the id exists in the database, but it’s acceptable. The worst case scenario is that the message will not be 100% accurate if the user specifically modifies the request to ask for a non-integer or non-existing collection ID.